File: frontend/src/types/collect-source.ts
export interface CollectSource {
  id: number;
  name: string;
  source_type: string;
  url: string;
  cron_expr: string;
  parser: string;
  is_enabled: boolean;
  last_run_at: string | null;
  last_status: string | null;
  created_at: string;
}

export interface CollectLog {
  id: number;
  source_id: number;
  status: string;
  items_found: number;
  items_new: number;
  error: string | null;
  started_at: string;
  finished_at: string | null;
}

export interface CollectRunResult {
  log_id: number;
  status: string;
  items_found: number;
  items_new: number;
}

export interface CollectSourceInput {
  name: string;
  source_type?: string;
  url: string;
  cron_expr?: string;
  parser?: string;
  is_enabled?: boolean;
}

File: backend/app/models/collect_source.py
from datetime import datetime
from typing import TYPE_CHECKING

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base

if TYPE_CHECKING:
    pass


class CollectSource(Base):
    __tablename__ = "collect_sources"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    source_type: Mapped[str] = mapped_column(String(50), nullable=False, default="rss")
    url: Mapped[str] = mapped_column(Text, nullable=False)
    cron_expr: Mapped[str] = mapped_column(String(50), default="0 8 * * *")
    parser: Mapped[str] = mapped_column(String(50), default="rss")
    is_enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    last_run_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))
    last_status: Mapped[str | None] = mapped_column(String(20))
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )

    logs: Mapped[list["CollectLog"]] = relationship(back_populates="source")


class CollectLog(Base):
    __tablename__ = "collect_logs"

    id: Mapped[int] = mapped_column(primary_key=True)
    source_id: Mapped[int] = mapped_column(ForeignKey("collect_sources.id", ondelete="CASCADE"))
    status: Mapped[str] = mapped_column(String(20), nullable=False)
    items_found: Mapped[int] = mapped_column(Integer, default=0)
    items_new: Mapped[int] = mapped_column(Integer, default=0)
    error: Mapped[str | None] = mapped_column(Text)
    started_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    finished_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True))

    source: Mapped[CollectSource] = relationship(back_populates="logs")
